libcxx/include/ccomplex

```diff
@@ -23,4 +23,12 @@
 #  pragma GCC system_header
 #endif
 
+#if _LIBCPP_STD_VER >= 17 && defined(__DEPRECATED) && __DEPRECATED && !defined(_LIBCPP_DISABLE_DEPRECATION_WARNINGS)
```

Comment:
What is the purpose of checking `__DEPRECATED` here?

Reply:
It's so people don't get deprecation warnings when specifying `-Wno-deprecated`.

Reply:
Should we instead set `_LIBCPP_DISABLE_DEPRECATION_WARNINGS` when `-Wno-deprecated` is passed?

Reply:
I'm not sure. You can have `-Wno-deprecated -Wdeprecated-declarations`, which disables deprecation warnings but keeps `[[deprecated]]` warnings. I'm not sure it makes a ton of sense to support this weird combination either though.